idaes/gas_solid_contactors/properties/oxygen_iron_OC_oxidation/gas_phase_thermo.py

```diff
@@ -0,0 +1,798 @@
+                    calculate_variable_from_constraint(
+                        blk[k].cp_mol_comp[j],
+                        blk[k].cp_shomate_eqn[j])
+
+                if hasattr(blk[k], "enthalpy_shomate_eqn"):
+                    calculate_variable_from_constraint(
+                            blk[k].enth_mol_comp[j],
+                            blk[k].enthalpy_shomate_eqn[j])
+
+        # Solve property block if non-empty
+        free_vars = 0
+        for k in blk.keys():
+            free_vars += number_unfixed_variables_in_activated_equalities(
+                blk[k])
+
+        if free_vars > 0:
+            with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
+                res = solve_indexed_blocks(opt, [blk], tee=slc.tee)
+        else:
+            res = ""
+        init_log.info_high("Initialization complete {}.".format(
+            idaeslog.condition(res))
+        )
+
+        # ---------------------------------------------------------------------
+        if state_vars_fixed is False:
+            if hold_state is True:
+                return flags
+            else:
+                blk.release_state(flags)
+
+    def release_state(blk, flags, outlvl=0):
+        """
+        Method to relase state variables fixed during initialization.
+        Keyword Arguments:
+            flags : dict containing information of which state variables
+                    were fixed during initialization, and should now be
+                    unfixed. This dict is returned by initialize if
+                    hold_state=True.
+            outlvl : sets output level of logging
+        """
+        if flags is None:
+            return
+
+        # Unfix state variables
+        revert_state_vars(blk, flags)
+
+        # Activate state variable related constraints
+        for k in blk.keys():
+            if blk[k].config.defined_state is False:
+                blk[k].sum_component_eqn.activate()
+
+        init_log = idaeslog.getInitLogger(blk.name, outlvl, tag="properties")
+        init_log.info_high('States released.')
+
+
+@declare_process_block_class("GasPhaseThermoStateBlock",
+                             block_class=_GasPhaseThermoStateBlock)
+class GasPhaseThermoStateBlockData(StateBlockData):
+    """
+    Property package for gas phase properties of methane combustion in CLC FR
+    """
+
+    def build(self):
+        """
+        Callable method for Block construction
+        """
+        super(GasPhaseThermoStateBlockData, self).build()
+
+        # Object reference for molecular weight if needed by CV1D
+        # Molecular weights
+        add_object_reference(self, "mw_comp",
+                             self.config.parameters.mw_comp)
+
+        """List the necessary state variable objects."""
+        self.flow_mol = Var(initialize=1.0,
+                            domain=Reals,
+                            doc='Component molar flowrate [mol/s]')
```

Comment:
You should start assigning units to variables now that we support them. Note that this will not immediately result in exceptions due to unit inconsistency, as these will only arise if you actually do a conversion or unit check.

However, we should start working towards full units support, and the first step is to start assigning units to `Vars` and `Params`.

Reply:
Yes, that's on my todo list. I won't have time to work on it in the next month. I'll setup a separate PR for it.